### ISSUE TITLE
Option to preserve page labels when merging

### ISSUE THREAD

**Issue by grr** [qpdf/qpdf#210]
I often have to reorder pages from the same document, but pages labels aren't preserved. The manual does document this deficiency and states a future version might correct this, but no ticket previously existed to track the issue.

**Comment by jberkenbilt** [comment]
Thanks for creating the ticket. There are several other tickets around similar issues. I'm going to create a new label for this and tag the related issues with it as well. There's a chance (no promises) that I may get to it this summer in conjunction with some closely related work.

**Comment by thfi** [comment]
I have very similar issue, albeit when extracting subsets of pages. To reproduce the issue, take the PDF file generated from the following LaTeX code:

```
\documentclass{article}
\usepackage{hyperref}
\begin{document}
\pagenumbering{roman}
This is page i
\clearpage
This is page ii
\clearpage
\pagenumbering{arabic}
This is page 1
\clearpage
This is page 2
\end{document}
```
To generate the PDF file, run `pdflatex page-numbers.tex`. The four pages of the PDF file are labeled/numbered `i`, `ii`, `1`, and `2`. Let's extract the first and the last page:
```
qpdf page-numbers.pdf  --pages page-numbers.pdf 1,4 -- page-numbers-selection.pdf
```
The resulting PDF file contains indeed the first and last page, but the last page's label is now `ii` instead of `2`.

**Comment by jberkenbilt** [comment]
Yup, thanks for the example. I can use this in testing once the feature is implemented.

**Comment by jberkenbilt** [comment]
This is fixed on master and will be included in qpdf 8.3.0, which should be out by the end of this week.
